Unable to create OWASP Core Rule Set

Hi,

When running `make modsec_crs_regression_tests_convert` it returns the following:

```
mkdir testcases/modsec-crs-false-pos
mv testcases/modsec-crs/fp_* testcases/modsec-crs-false-pos/
mv: cannot stat 'testcases/modsec-crs/fp_*': No such file or directory
make: *** [Makefile:40: modsec_crs_regression_tests_convert] Error 1
```

I've added some debugging to the modsec_regression_testset_convertor.rb file to get some details of where it's breaking and here's a snippet from the debug logs:

```
D, [2025-03-18T12:25:57.988145 #57078] DEBUG -- : Getting rules paranoia level for: REQUEST-934-APPLICATION-ATTACK-GENERIC
D, [2025-03-18T12:25:57.988635 #57078] DEBUG -- : Found rule ID: 934011, Paranoia Level: 1
D, [2025-03-18T12:25:57.988707 #57078] DEBUG -- : Found rule ID: 934012, Paranoia Level: 1
D, [2025-03-18T12:25:57.988855 #57078] DEBUG -- : Found rule ID: 934100, Paranoia Level: 1
D, [2025-03-18T12:25:57.988956 #57078] DEBUG -- : Found rule ID: 934110, Paranoia Level: 1
D, [2025-03-18T12:25:57.989018 #57078] DEBUG -- : Found rule ID: 934130, Paranoia Level: 1
D, [2025-03-18T12:25:57.989100 #57078] DEBUG -- : Found rule ID: 934150, Paranoia Level: 1
D, [2025-03-18T12:25:57.989197 #57078] DEBUG -- : Found rule ID: 934160, Paranoia Level: 1
D, [2025-03-18T12:25:57.989292 #57078] DEBUG -- : Found rule ID: 934170, Paranoia Level: 1
D, [2025-03-18T12:25:57.989381 #57078] DEBUG -- : Found rule ID: 934013, Paranoia Level: 2
D, [2025-03-18T12:25:57.989430 #57078] DEBUG -- : Found rule ID: 934014, Paranoia Level: 2
D, [2025-03-18T12:25:57.989457 #57078] DEBUG -- : Found rule ID: 934101, Paranoia Level: 2
D, [2025-03-18T12:25:57.989518 #57078] DEBUG -- : Found rule ID: 934120, Paranoia Level: 2
D, [2025-03-18T12:25:57.989559 #57078] DEBUG -- : Found rule ID: 934140, Paranoia Level: 2
D, [2025-03-18T12:25:57.989577 #57078] DEBUG -- : Found rule ID: 934015, Paranoia Level: 3
D, [2025-03-18T12:25:57.989627 #57078] DEBUG -- : Found rule ID: 934016, Paranoia Level: 3
D, [2025-03-18T12:25:57.989666 #57078] DEBUG -- : Found rule ID: 934017, Paranoia Level: 4
D, [2025-03-18T12:25:57.989769 #57078] DEBUG -- : Found rule ID: 934018, Paranoia Level: 4
D, [2025-03-18T12:25:58.019033 #57078] DEBUG -- : Separating negative and positive tests
D, [2025-03-18T12:25:58.019359 #57078] DEBUG -- : Converting and saving GET test cases for: modsec-crs-generic
W, [2025-03-18T12:25:58.019404 #57078]  WARN -- : No test cases found for: modsec-crs-generic
D, [2025-03-18T12:25:58.019490 #57078] DEBUG -- : Converting and saving POST test cases for: modsec-crs-generic
W, [2025-03-18T12:25:58.019574 #57078]  WARN -- : No test cases found for: modsec-crs-generic
D, [2025-03-18T12:25:58.019658 #57078] DEBUG -- : Converting and saving GET test cases for: fp_get_modsec_crs
W, [2025-03-18T12:25:58.019688 #57078]  WARN -- : No test cases found for: fp_get_modsec_crs
D, [2025-03-18T12:25:58.019745 #57078] DEBUG -- : Converting and saving POST test cases for: fp_post_modsec_crs
W, [2025-03-18T12:25:58.019825 #57078]  WARN -- : No test cases found for: fp_post_modsec_crs
I, [2025-03-18T12:25:58.019878 #57078]  INFO -- : Processing completed.


There has been no other changes to the rb file, just straight out the box (besides adding in some logger code to debug)

--- a/.gitignore
+++ b/.gitignore
@@ -1,4 +1,5 @@
 .idea
+.tmp
 .DS_store
 /gotestwaf
 /main

--- a/misc/modsec_regression_testset_converter.rb
+++ b/misc/modsec_regression_testset_converter.rb
@@ -152,9 +152,9 @@ crs_testcases.each do |gtw_testcase_name, test_set|
     next unless file.match(/.yaml$/)
 
     parsed_yaml_data = YAML.safe_load(File.read(file))
-    next if parsed_yaml_data['meta']['enabled'] != true
 
-    rule_id = parsed_yaml_data['meta']['name'].split('.')[0]
+    rule_id = parsed_yaml_data['rule_id']
+
     next if paranoia_level[rule_id].nil? || paranoia_level[rule_id] >= max_paranoia_level
     parsed_yaml_data['tests'].each { |t| current_test_set_data << t }
   end

--- a/misc/modsec_regression_testset_converter.rb
+++ b/misc/modsec_regression_testset_converter.rb
@@ -22,7 +22,7 @@ def separate_negative_and_positive_tests(tests)
   result = { 'positive' => [], 'false-positive' => [] }
 
   tests.each do |t|
-    if t['stages'][0]['stage']['output'].keys.include? 'no_log_contains' and t['stages'][0]['stage']['output']['no_log_contains'].match(/^id/i)
+    if t['stages'][0]['output']['log'].keys.include? 'no_expect_ids' and t['stages'][0]['output']['log']['no_expect_ids'].length() == 1
       result['false-positive'] << t
     else
       result['positive'] << t
@@ -43,7 +43,7 @@ def get_rules_paranoia_level(test_set)
     paranoia = line.match(/DETECTION_PARANOIA_LEVEL "@.*(\d+)"/i)[1] if line.match(/DETECTION_PARANOIA_LEVEL "@.*\d+"/i)
     next if paranoia.nil? || id.nil?
 
-    result[id] = paranoia.to_i
+    result[id.to_i] = paranoia.to_i
   end
 
   result
@@ -51,16 +51,16 @@ end
 
 def split_tests_by_req_method(tests)
   result = []
-  test_title = tests['test_title']
+  test_id = tests['test_id']
 
   tests['stages'].each do |stage|
-    case stage['stage']['input']['method']
+    case stage['input']['method']
     when nil
-      result << parse_get_requests(stage['stage']['input'], test_title)
+      result << parse_get_requests(stage['input'], test_id)
     when 'POST'
-      result << parse_post_requests(stage['stage']['input'], test_title)
+      result << parse_post_requests(stage['input'], test_id)
     else
-      result << parse_get_requests(stage['stage']['input'], test_title)
+      result << parse_get_requests(stage['input'], test_id)
     end
   end
 

